exe/cli/cmd/ffs_push.go

```diff
@@ -33,21 +41,47 @@ var ffsAddCmd = &cobra.Command{
 			Fatal(errors.New("you must provide a cid"))
 		}
 
-		token := viper.GetString("token")
+		c, err := cid.Parse(args[0])
+		checkErr(err)
+
+		configPath := viper.GetString("config")
 
-		if token == "" {
-			Fatal(errors.New("add requires token"))
+		var reader io.Reader
+		if len(configPath) > 0 {
+			file, err := os.Open(configPath)
+			defer file.Close()
+			reader = file
+			checkErr(err)
+		} else {
+			stat, _ := os.Stdin.Stat()
+			// stdin is being piped in (not being read from terminal)
+			if (stat.Mode() & os.ModeCharDevice) == 0 {
+				reader = cmd.InOrStdin()
+			}
 		}
```

Comment:
You can provide a config via path to a file containing json, or by piping the json to stdin.